src/mam4xx/mo_sethet.hpp

```diff
@@ -295,21 +295,22 @@ void sethet_detail(
   // 'H2O2','H2SO4','SO2'.  Options for other species are then removed
   //-----------------------------------------------------------------
 
-  Kokkos::parallel_for(Kokkos::TeamThreadRange(team, local_pver), [&](int kk) {
-    Kokkos::parallel_for(
-        Kokkos::ThreadVectorRange(team, gas_pcnst), [&](int mm) {
+  Kokkos::parallel_for(
+      Kokkos::ThreadVectorRange(team, local_pver), [&](int kk) {
+        for (int mm = 0; mm < gas_pcnst; ++mm) {
           het_rates(kk, mm) = 0.0;
           tmp_hetrates[mm](kk) = 0.0; // initiate temporary array
-        });
-  });
+        }
+      });
 
-  for (int mm = 0; mm < gas_wetdep_cnt; mm++) {
-    int mm2 = wetdep_map[mm];
-    if (mm2 > 0) {
-      Kokkos::parallel_for(Kokkos::TeamVectorRange(team, local_pver),
-                           [&](int kk) { het_rates(kk, mm2) = MISSING; });
-    }
-  }
+  Kokkos::parallel_for(Kokkos::ThreadVectorRange(team, local_pver),
+                       [&](int kk) {
+                         for (int mm = 0; mm < gas_wetdep_cnt; mm++) {
+                           const int mm2 = wetdep_map[mm];
+                           if (mm2 > 0)
```

Comment:
@mjs271 and I discovered yesterday that this needs to be `>=` to account for fortran -> C++ indexing

Reply:
Done.....